maxminddb/reader.py

```diff
@@ -38,30 +38,39 @@ def __init__(self, database, mode=MODE_AUTO):
         """Reader for the MaxMind DB file format
 
         Arguments:
-        database -- A path to a valid MaxMind DB file such as a GeoIP2
-                    database file.
+        database -- A path to a valid MaxMind DB file such as a GeoIP2 database
+                    file, or a file descriptor in the case of MODE_FD.
         mode -- mode to open the database with. Valid mode are:
             * MODE_MMAP - read from memory map.
             * MODE_FILE - read database as standard file.
             * MODE_MEMORY - load database into memory.
             * MODE_AUTO - tries MODE_MMAP and then MODE_FILE. Default.
+            * MODE_FD - the param passed via database is a file descriptor, not
+                        a path. This mode implies MODE_MEMORY.
         """
         if (mode == MODE_AUTO and mmap) or mode == MODE_MMAP:
             with open(database, 'rb') as db_file:
                 self._buffer = mmap.mmap(
                     db_file.fileno(), 0, access=mmap.ACCESS_READ)
                 self._buffer_size = self._buffer.size()
+            filename = database
         elif mode in (MODE_AUTO, MODE_FILE):
             self._buffer = FileBuffer(database)
             self._buffer_size = self._buffer.size()
+            filename = database
         elif mode == MODE_MEMORY:
             with open(database, 'rb') as db_file:
                 self._buffer = db_file.read()
                 self._buffer_size = len(self._buffer)
+            filename = database
+        elif mode == MODE_FD:
+            self._buffer = database.read()
+            self._buffer_size = len(self._buffer)
+            filename = database.name
```

Comment:
Of course, [and done](https://github.com/maxmind/MaxMind-DB-Reader-python/pull/33/commits/05c95bd2ff7c9bc87ecee9eb9b9092ba26c23c02)! Since the various modes are briefly described in numerous places/files, I elected to add this bit of documentation to `README.rst`. Does this seem sufficient?

Also, if this PR gets merged, I've got another small one for the [GeoIP2-python module](https://github.com/maxmind/GeoIP2-python) to enable this feature in that wrapper as well.

Reply:
Yeah, that seems fine for now. Thanks!